kernel/src/main/java/org/kframework/kompile/Kompile.java

```diff
@@ -437,6 +438,8 @@ public void structuralChecks(scala.collection.Set<Module> modules, Module mainMo
 
         stream(modules).forEach(m -> stream(m.localSentences()).forEach(new CheckHOLE(errors, m)::check));
 
+        stream(modules).forEach(m -> stream(m.localSentences()).forEach(new CheckTokens(errors, m, isSymbolic && isKast)::check));
```

Comment:
I think this check `isSymbolic && isKast` can be done before even entering this line.
So we would avoid this entire step instead of looking at each sentence and decide if we need to do something.